ocamltest/run_unix.c

```diff
@@ -0,0 +1,273 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*             Sébastien Hinderer, projet Gallium, INRIA Paris            */
+/*                                                                        */
+/*   Copyright 2016 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+/* Run programs with rediretions and timeouts under Unix */
+
+#include <stdio.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <stdarg.h>
+#include <signal.h>
+
+#include "run.h"
+#include "run_common.h"
+
+#define COREFILENAME "core"
+
+static volatile int timeout_expired = 0;
+
+#define error(msg, ...) \
+error_with_location(__FILE__, __LINE__, settings, msg, ## __VA_ARGS__)
+
+/*
+  Note: the ## __VA_ARGS__ construct is gcc specific.
```

Comment:
I presume it also works in clang, but people with more exotic compilers (icc, Sun) will be sad.